geth/transactions/factory.go

```diff
@@ -0,0 +1,188 @@
+package transactions
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/rpc"
+	"github.com/status-im/status-go/sign"
+)
+
+const (
+	// sendTransactionTimeout defines how many seconds to wait before returning result in sentTransaction().
+	sendTransactionTimeout = 300 * time.Second
+	rpcCallTimeout         = time.Minute
+
+	defaultGas = 90000
+)
+
+// Factory validates, signs and propagates transactions to the Ethereum network.
+type Factory struct {
+	signRequests      *sign.PendingRequests
+	ethTxClient       EthTransactor
+	completionTimeout time.Duration
+	rpcCallTimeout    time.Duration
+	networkID         uint64
+
+	addrLock   *AddrLocker
+	localNonce sync.Map
+	log        log.Logger
+}
+
+// NewFactory returns a new Manager.
+func NewFactory(signRequests *sign.PendingRequests) *Factory {
+	return &Factory{
+		signRequests:      signRequests,
+		addrLock:          &AddrLocker{},
+		completionTimeout: sendTransactionTimeout,
```

Comment:
Maybe use the same name here, either `completionTImeout` or `sendTransactionTimeout`?